### ISSUE TITLE
Caught an exception on ProxyToServerConnection

### ISSUE THREAD

**Issue by madoni** [lightbody/browsermob-proxy#478]
@jekh 

There are instances that I see this message in the log output for `browsermob`.

```
11:46:13.050 [LittleProxy-0-ProxyToServerWorker-5] ERROR org.littleshoot.proxy.impl.ProxyToServerConnection - (AWAITING_INITIAL) [id: 0x2748bd19, L:/10.212.30.79:54450 - R:<corporate proxy>/<ip>:8099]: Caught an exception on ProxyToServerConnection
io.netty.handler.codec.DecoderException: java.lang.NullPointerException
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:418)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:292)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:278)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:962)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:528)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:399)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:371)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException: null
	at java.lang.System.arraycopy(Native Method)
	at com.sun.crypto.provider.GCTR.reset(GCTR.java:125)
	at com.sun.crypto.provider.GCTR.doFinal(GCTR.java:116)
	at com.sun.crypto.provider.GaloisCounterMode.doLastBlock(GaloisCounterMode.java:343)
	at com.sun.crypto.provider.GaloisCounterMode.decryptFinal(GaloisCounterMode.java:511)
	at com.sun.crypto.provider.CipherCore.finalNoPadding(CipherCore.java:1023)
	at com.sun.crypto.provider.CipherCore.doFinal(CipherCore.java:960)
	at com.sun.crypto.provider.AESCipher.engineDoFinal(AESCipher.java:479)
	at javax.crypto.CipherSpi.bufferCrypt(CipherSpi.java:830)
	at javax.crypto.CipherSpi.engineDoFinal(CipherSpi.java:730)
	at javax.crypto.Cipher.doFinal(Cipher.java:2416)
	at sun.security.ssl.CipherBox.decrypt(Unknown Source)
	at sun.security.ssl.EngineInputRecord.decrypt(Unknown Source)
	at sun.security.ssl.SSLEngineImpl.readRecord(Unknown Source)
	at sun.security.ssl.SSLEngineImpl.readNetRecord(Unknown Source)
	at sun.security.ssl.SSLEngineImpl.unwrap(Unknown Source)
	at javax.net.ssl.SSLEngine.unwrap(Unknown Source)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1098)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:970)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:904)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:387)
	... 11 common frames omitted
```

I am running `v1.1.0-beta-bmp-12` in conjunction with `browsermob v2.1.0`.

The chained proxy is the corporate proxy which I am assuming is TCP and TLS (how can I check this information?)
Looking at the project dependencies, I am using `netty-all v4.0.36` which is a `browsermob-core v2.1.0` dependency.

Thanks once again!

**Comment by jekh** [comment]
@madoni - that is something you don't see every day, a NPE deep in the Sun crypto library. It appears to be an instance of this JDK bug: https://bugs.openjdk.java.net/browse/JDK-8067755. Bizarrely, that issue was closed as a duplicate of JDK-8049855 and JDK-8150150, which were both closed as duplicates of each other! So this appears to be a known issue, but I cannot find any status on it as the bug reports were improperly closed.

Can you try on the latest version of Java, 8u92? It may already be fixed.

**Comment by madoni** [comment]
Ok... this makes sense.
This application is compiled using `jdk1.8.0_92`.
I use `java version "1.8.0_92", Java(TM) SE Runtime Environment (build 1.8.0_92-b14)` and I don't see this error running a particular test.
Another developer on the team uses `java version "1.8.0_11", Java(TM) SE Runtime Environment (build 1.8.0_11-b12)` and sees this error while running the same test!!
I'll see if an updating the jre on the other system sorts it out... I think it should.

Thanks for looking into this! Appreciate all the support!
